scripts/superdesk-authoring/metadata/metadata.js

```diff
@@ -700,17 +700,25 @@ function MetaTermsDirective(metadata, $filter, $timeout) {
             function filterSelected(terms) {
                 var selected = {};
                 angular.forEach(scope.item[scope.field], function(term) {
-                    selected[term[scope.uniqueField]] = 1;
+                    if (term) {
+                        selected[term[scope.uniqueField]] = 1;
+                    }
                 });
 
                 return _.filter(terms, function(term) {
-                    return !selected[term[scope.uniqueField]];
+                    return term && !selected[term[scope.uniqueField]];
                 });
             }
 
             function addTerm(term) {
+                if (!term) {
+                    return;
+                }
+
                 // Only select terms that are not already selected
-                if (!_.find(scope.item[scope.field], function(i) {return i[scope.uniqueField] === term[scope.uniqueField];})) {
+                if (!_.find(scope.item[scope.field], function(i) {
```

Comment:
It would be easier to figure this logic out, imo, if we would separate it into a function. One possible example could be:
```js
let uf = scope.uniqueField;
let isSelected = _.find(scope.item[scope.field], i => i && i[uf] === term[uf];

if (isSelected) {
    return;
}

// ...
```
This is just an alternative. It also reduced indentation. But, if you feel strongly about yours, I'm happy to keep it.